pkg/volume/csi/shared_informer_hook.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// SharedInformerHook provides a way to dynamically watch objects via a SharedInformer.
```

Comment:
I'm not sure if it's necessary to add `SharedInformerHook`.
You can create informer to watch object of specific name by use `tweakListOptions` to add [fieldSelector](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L339). IIRC, the fieldSelector for object name is `metadata.name`.

Reply:
@caesarxuchao Can I dynamically modify this list of options while the SharedInformer is already running?

I want to allow the CSI plugin's SharedInformer to run, and while it's running I want to dynamically start and stop watching for many objects by name. I need a way of taking the events for a specific tracked object and funneling it to the right channel for that one object. I'm unsure if `tweakListOptions` would enable me to produce this kind of behavior.

Reply:
You are right, that's not supported by the shared Informer. Thanks for the explanation!